src/loader/BlendLoader.py

```diff
@@ -73,28 +122,115 @@ def run(self):
         else:
             entities = None
 
-        previously_loaded_objects = set(bpy.context.scene.objects)
+        data_block_name = load_from.strip("/")
 
-        with bpy.data.libraries.load(path) as (data_from, data_to):
+        with bpy.data.libraries.load(path) as (blend_file_data, _):
             # check if defined ID is supported
-            if load_from in self.known_datablock_names.keys():
+            if data_block_name in self.known_datablock_names:
+                attr_name = self._find_datablock_name_match_in_blendfile(
+                    blend_file_data, data_block_name)
+
                 # if some regex was specified, get corresponding matching entity's names
                 if entities is not None:
-                    entities_to_load = [item for item in getattr(data_from, self.known_datablock_names[load_from])
-                                        if re.fullmatch(entities, item) is not None]
+                    entities_to_load = [
+                        item for item in getattr(blend_file_data, attr_name)
+                        if re.fullmatch(entities, item) is not None
+                    ]
+
                 # get all entity's names if not
                 else:
-                    entities_to_load = getattr(data_from, self.known_datablock_names[load_from])
+                    entities_to_load = getattr(blend_file_data, attr_name)
                 # load entities
                 for entity_to_load in entities_to_load:
-                    bpy.ops.wm.append(filepath=os.path.join(path, load_from, entity_to_load),
-                                      filename=entity_to_load,
-                                      directory=os.path.join(path + load_from))
+
+                    # remove the earlier existing resource with same name
+                    if entity_to_load in bpy.data.objects:
+                        bpy.data.objects.remove(
+                            bpy.data.objects[entity_to_load], do_unlink=True)
+
+                    bpy.ops.wm.append(
+                        filepath=os.path.join(path, data_block_name,entity_to_load),
+                        filename=entity_to_load,
+                        directory=os.path.join(path, data_block_name))
+
+                    added_resource = getattr(bpy.data, attr_name)[entity_to_load]
+
+                    if hasattr(added_resource, 'type') and added_resource.type == 'CAMERA':
+                        bpy.context.scene.collection.objects.link(added_resource)
```

Comment:
We should also remove the last camera. The main init creates one, but it is not used anymore after this.